File: aoc2024/clojure-solutions/src/clojure_solutions/day01.clj
(ns clojure-solutions.day01
  (:require [clojure.string :as str]
            [clojure-aoc-util.util :as util])
  (:gen-class))

(defn- parse []
  (->> (slurp "../inputs/day01.txt")
       str/split-lines
       (map (comp (partial map read-string)
                  #(str/split % #" +")))
       util/transpose
       (map sort)))

(defn -main [& _args]
  (let [[a b] (parse)
        freqs (frequencies b)]
    (util/print-day
     1
     (reduce + (map (comp abs -) a b))
     (reduce + (for [x a] (* x (get freqs x 0)))))))

File: aoc2019/c_solutions/day01.c
#include "util.h"

int main() {
    sz n=0,o=0,t=0,*ns=read_ints("../inputs/day01.txt", &n);
    DO(n, _(o+=ns[i]/3-2;
            int tmp=0,el=ns[i];while(el>0){el=el/3-2;tmp+=el;}t+=tmp;));
    CHECK(o, 3334282);
    CHECK(t, 4998467);
    free(ns);
}
